internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,255 @@
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+type rule = aggregationv1.KeyerConfiguration_Fragment_Rule
+type matchPredicate = aggregationv1.MatchPredicate
+type repeatedResultPredicate = aggregationv1.ResultPredicate_RepeatedResultPredicate
+
+// Mapper defines the interface that Maps an incoming request to an aggregation key
+type Mapper interface {
+	GetKey(node core.Node, typeURL string) (string, error)
+}
+
+type mapper struct {
+	config aggregationv1.KeyerConfiguration
+}
+
+// NewMapper construts a concrete implementation for the Mapper interface
+func NewMapper(config aggregationv1.KeyerConfiguration) Mapper {
+	return &mapper{
+		config: config,
+	}
+}
+
+// GetKey converts a request into an aggregated key
+func (mapper *mapper) GetKey(node core.Node, typeURL string) (string, error) {
+	for _, fragment := range mapper.config.GetFragments() {
+		fragmentrules := fragment.GetRules()
+		for _, fragmentrule := range fragmentrules {
+			matchpredicate := fragmentrule.GetMatch()
+			isMatch, err := isMatchPredicate(matchpredicate, node, typeURL)
+			if err != nil {
+				return "", err
+			}
+			if isMatch {
+				return getResult(fragmentrule, node)
+			}
+		}
```

Comment:
I think we're missing appending of the other fragments.
i.e. if the keyer configuration had 3 fragments, we're returning
`<fragment_result_1>` rather than,
`<fragment_result_1>_<fragment_result_2>_<fragment_result_3>`

We still pick the first rule match, but need to consider the other fragments as well.

Reply:
right. Fixing this today